packages/cli/yaml/validator/src/ComplexQueryParamTypeDetector.ts

```diff
@@ -27,12 +27,15 @@ export class ComplexQueryParamTypeDetector {
         this.typeResolver = new TypeResolverImpl(workspace);
     }
 
-    public isTypeComplex(type: string, ruleRunnerArgs: RuleRunnerArgs<DefinitionFileSchema>): boolean | undefined {
+    public async isTypeComplex(
+        type: string,
+        ruleRunnerArgs: RuleRunnerArgs<DefinitionFileSchema>
+    ): Promise<boolean | undefined> {
         const file = constructFernFileContext({
             relativeFilepath: ruleRunnerArgs.relativeFilepath,
             definitionFile: ruleRunnerArgs.contents,
             casingsGenerator: CASINGS_GENERATOR,
-            rootApiFile: this.workspace.definition.rootApiFile.contents
+            rootApiFile: (await this.workspace.getDefinition()).rootApiFile.contents
```

Comment:
Can we make `getDefinition` stateful so that once it's computed for a language we just keep it around? otherwise this change is going to lead to potentially a lot of slow down in different places

Reply:
yeah

Reply:
implemented and pushing in a sec